README.md

```diff
@@ -0,0 +1,87 @@
+
+Let's say we want to write a program that makes a sandwhich, a peanbut butter sandwhich. What are the steps of making a sandwhich? Well, we all know that:
+
+```
+1) Prepare workspace.  
+2) Gather you ingredients:   
+      * White Bread
+      * Peanut Butter
+      * Jelly
+3) Put to slices of bread on workspace.
+4) Spread peanut butter on one slice.
+5) Spread jam on the other slice.
+6) Slap those slices together.
+8) Boom! Peanut Butter Sandwich.
+```
+
+![](http://ezmiller.s3.amazonaws.com/public/flatiron-imgs/peanutbutter.gif)
+
+So there we have an algorithm, which can be performed one-step-after-another, i.e., synchronously. Pretty straight-forward. Now let's complicate things.
+
+Let's say that we want our algorithm to use freshly baked bread, and that we also want the algorithm to be able to the handle scenario in which there's no peanut butter or jelly available. 
+Now we have a problem. Because suddenly step #1 above is going to take time. Worse, it may even take an indefinite amount of time because we don't know *when exactly* the visit to the grocery store will be completed or when the bread will be baked. Anything can happen really. The bread could burn. Our regular grocer, remarkably, could be out of peanut butter! 
+
+Suddenly our nice algorithmic recipe above is no longer synchronous; it's **A**sychronous! We cannnot proceed with step #3 until all of the processes involved in step #2 have finished. Using our more general definition of async above, here we have a situation in which we want part of our program (i.e. the preparation of the work space and the intial gathering of ingredients) to run immediately, and then later, at some undetermined moment when we have everything, we want to complete the task (i.e. make the sandwich).
+
+This, then, is an asynchronous procedure. The key is that **we don't know when exactly in the future "later" will be; it's just *sometime* later.**
+
+## Callbacks to the Rescue
+
+Now that we've accomplished our first objective -- learning how to explain what an asychronous process is -- let's examine the most basic and common method for handling asynchrony using a pattern known as a "callback." Since you've probably already used jQuery quite a bit, it is very likely that you've already written a few callback functions, perhaps without knowing it. If so, maybe something of this will feel familiar!
+
+Before we jump into an example of a callback, let's imagine a more real-world async example. Instead of making a sandwich, let's think about fetching search results from a server and displaying them to a user. All we need to  do is the following:
+
+```
+1. Send the search terms to the server that will fetch our results.
+3. Display the results.
```

Comment:
I think this is inconsequential in markdown, but just in case -- looks like there's no step 2.